### PR TITLE
[RLM-712] Clean glance cache before leap

### PR DESCRIPTION
**Pull request by Frank-ZhangXin**
This action will wipe all glance image cache before leaping to prevent
the container too full to achieve upgrading.

### CODE REVIEW

File: playbooks/glance-cache-cleanup.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      file:
+        state: absent
+        path: "/var/lib/glance/cache/"
+
+    - name: Generate cache directory
+      file:
+        state: directory
+        path: "/var/lib/glance/cache/"
+
+    - name: Restore glance-api
```
Comment (antonym) [inline playbooks/glance-cache-cleanup.yml]: Might be a good idea to start glance-registry back up first since it's the datastore API talks to.

**Comment by antonym** [comment]
Looks like gate is failing with: ERROR: /opt/rpc-openstack/openstack-ansible/playbooks/inventory/env.d/aodh.yml:5: Invalid ini entry: aodh_all - need more than 1 value to unpack

Usually this means the inventory doesn't match the proper version of ansible.  Maybe this needs to run earlier before anything is modified or upgraded.

File: playbooks/glance-cache-cleanup.yml (hunk 1 above)
Comment (Frank-ZhangXin) [reply playbooks/glance-cache-cleanup.yml]: cool

**Comment by Frank-ZhangXin** [comment]
@antonym I moved timing of glance cache wipe to beginning of pre_leap script, hope it's soon enough to avoid aodh error on gate

**Comment by antonym** [comment]
I think it needs to run before this point here:

https://github.com/rcbops/rpc-upgrades/blob/master/scripts/ubuntu14-leapfrog.sh#L98

Because after rpc-openstack of the target version is checked out, it contains the inventory for Newton but is still running an older version of Ansible (1.9 or so) that does not understand how to parse the inventory.  

The other option would be to checkout rpc-openstack of the current version you are leaping from so that it can process the inventory right and then move it back to the target.

**Comment by Frank-ZhangXin** [comment]
@antonym I moved the playbook into ubuntu14, hope it would work.

**Comment by Frank-ZhangXin** [comment]
Still got error on gate:
ERROR: /opt/rpc-openstack/openstack-ansible/playbooks/inventory/env.d/aodh.yml:5: Invalid ini entry: aodh_all - need more than 1 value to unpack

File: scripts/ubuntu14-leapfrog.sh [hunk 2]
```diff
@@ -85,6 +85,11 @@ else
   log "rpc-preflight-check" "skipped"
 fi
 
+# Glance cache cleanup
+pushd /opt/rpc-upgrades/playbooks/
+    openstack-ansible glance-cache-cleanup.yml
+pop
```
Comment (antonym) [inline scripts/ubuntu14-leapfrog.sh]: Resolved the conflict, but this should be popd

Comment (Frank-ZhangXin) [reply scripts/ubuntu14-leapfrog.sh]: Yeah, fixing it now

**Comment by antonym** [comment]
recheck_cit_trusty_aio_swift_liberty_to_newton_leap

File: playbooks/glance-cache-cleanup.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    - name: Generate cache directory
+      file:
+        state: directory
+        path: "/var/lib/glance/cache/"
+
+    - name: Restore glance-registry
+      service:
+        name: glance-registry
+        state: started
+        patter: glance-registry
```
Comment (antonym) [inline playbooks/glance-cache-cleanup.yml]: Looks like its running in gate now, but two cases where pattern is mispelled is stopping it from running.

File: playbooks/glance-cache-cleanup.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+    - name: Restore glance-api
+      service:
+        name: glance-api
+        state: started
+        patter: glance-api
```
Comment (antonym) [inline playbooks/glance-cache-cleanup.yml]: pattern (sp)

**Comment by antonym** [comment]
Passing in Liberty now, will merge once kilo is good.

**Comment by antonym** [comment]
recheck_trusty_aio_swift_kilo_to_newton_leap

**Pull request merged**